tracer/src/Datadog.Trace/OTelMetrics/OtlpMetricsSerializer.cs

```diff
@@ -252,28 +252,31 @@ private byte[] SerializeSum(MetricPoint metric)
             WriteVarInt(writer, dataPointData.Length);
             writer.Write(dataPointData);
 
-            int temporality;
+            // Determine temporality and monotonicity based on instrument type
+            AggregationTemporality temporality;
             bool isMonotonic;
 
-            if (metric.InstrumentName.Contains("upDownCounter"))
+            if (metric.InstrumentType is InstrumentType.UpDownCounter or InstrumentType.ObservableUpDownCounter)
             {
-                temporality = 2;
+                // UpDownCounter is always cumulative and non-monotonic
+                temporality = AggregationTemporality.Cumulative;
                 isMonotonic = false;
             }
             else
             {
+                // Counter and ObservableCounter use configured temporality and are monotonic
                 temporality = _settings.OtlpMetricsTemporalityPreference switch
                 {
-                    OtlpTemporality.Delta => 1,
-                    OtlpTemporality.LowMemory => 1,
-                    OtlpTemporality.Cumulative => 2,
-                    _ => 1
+                    OtlpTemporality.Delta => AggregationTemporality.Delta,
```

Comment:
😍 Looks great! The only other improvement I can think of is renaming the class name for the `OtlpTemporality` enum to make the distinction a little clearer that this one is the "Preference". For people looking at this code for the first time, it might not be obvious what the difference is between the right and left side of the switch expressions. Can definitely do this as a follow-up if you want to pursue it